TeamCode/src/main/java/org/firstinspires/ftc/teamcode/KushaalDriverOpMode.java

```diff
@@ -110,47 +107,34 @@ public void runOpMode() throws InterruptedException {
 
         double speed = 0.75;
         double strafe = 0;
-
+        double max;
         // run until the end of the match (driver presses STOP)
         while (opModeIsActive()) {
-            double max;
 
-            if (speed < 0){
+            if (speed < 0) {
```

Comment:
Hi Samvith, the capping of `speed` to zero and to 100 is happening here, but later code detects `gamepad1.dpad` and increase and decrease speed accordingly, which would result in speed below 0 and above 100, and passing those out-of-bound values to the the motor power calculation. 

I think you should consider creating a method called `increase_speed `and `decrease_speed`, which will cap the `this.speed` value to 0 and 100. And then a method called `get_speed` to get the speed.